### PR TITLE
separate deposit into two instructions: deposit, deposit_and_call

### PR DESCRIPTION
**Pull request by brewmaster012**
deposit only takes amount and receiver address

deposit_and_call takes amount, receiver address (should be a contract), and then the message (calldata) into that contract call

closes: #18

### CODE REVIEW

**Review by ws4charlie — APPROVED** [review]
LGTM

File: programs/protocol-contracts-solana/src/lib.rs [hunk 1]
```diff
@@ -101,48 +97,41 @@ pub mod gateway {
...
         amount: u64,
-        memo: Vec<u8>,
+        receiver: [u8; 20],
+        message: Vec<u8>,
     ) -> Result<()> {
-        require!(memo.len() >= 20, Errors::MemoLengthTooShort);
-        require!(memo.len() <= 512, Errors::MemoLengthExceeded);
-        let token = &ctx.accounts.token_program;
-        let from = &ctx.accounts.from;
+        require!(message.len() <= 512, Errors::MemoLengthExceeded);
```
Comment (lumtis) [inline programs/protocol-contracts-solana/src/lib.rs:115]: Can we increase this limit?

**Review by lumtis — APPROVED** [review]

File: programs/protocol-contracts-solana/src/lib.rs (hunk 1 above)
Comment (brewmaster012) [reply programs/protocol-contracts-solana/src/lib.rs:115]: to 4096?

Comment (brewmaster012) [reply programs/protocol-contracts-solana/src/lib.rs:115]: adjusted to 1024 there is not reason to go beyond 1024
as that's the limit of transaction size on solana.

**Pull request merged**